src/torchcodec/decoders/_video_decoder.py

```diff
@@ -379,3 +394,40 @@ def _get_and_validate_stream_metadata(
         end_stream_seconds,
         num_frames,
     )
+
+
+def read_custom_frame_mappings(
+    custom_frame_mappings: Union[bytes, bytearray, str]
+) -> tuple[Tensor, Tensor, Tensor]:
+    try:
+        if hasattr(custom_frame_mappings, "read"):
+            input_data = json.load(custom_frame_mappings)
+        else:
+            input_data = json.loads(custom_frame_mappings)
+    except json.JSONDecodeError:
+        raise ValueError(
+            "Invalid custom frame mappings. "
+            "It should be a valid JSON string or a JSON file object."
+        )
+    # These keys are prefixed with "pkt_" in ffmpeg 4 and ffmpeg 5
+    pts_key = "pkt_pts" if "pts" not in input_data["frames"][0] else "pts"
```

Comment:
Nit, but it would be helpful to users to do explicit error checks on our assumptions about the data:

1. Assert that `input_data` is non-empty.
2. Assert that `"frames"` is a valid key.
3. Assert that one of the valid pts and duration keys exist. (Rather than assuming it must be the other one.)

If any of the above isn't true, we'll still fail, but it will just be with `KeyError` exceptions that the cause of may not be obvious to users.